File: ClickDigitalBackend/src/main/java/rulemanager/models/Triggergroup.java
package rulemanager.models;

import com.fasterxml.jackson.annotation.JsonAutoDetect;

/**

 * An object of this class represents a internal triggergroup.
 */
@JsonAutoDetect(fieldVisibility = JsonAutoDetect.Visibility.ANY)
public class Triggergroup {
    public String name = "";
    public int ID = 0;
    public String operator = "";
    public int leftchild = 0;
    public int rightchild = 0;
    public Trigger trigger = null;
    public String userId;
    public int triggerID;
    public String triggerclass;
    public String projectID;

    public Triggergroup(){}
}

File: ClickDigitalFrontend/src/app/models/frontend/rule_module/triggergroup.ts
import { Trigger } from './trigger';

// Triggergroup template for matching with JSON Object  from frontend received by backend
export class  Triggergroup {

  ID: number;
  name = 'default-name'; // for save value is needed
  operator = ''; // OR or AND
  leftchild = 0;
  rightchild = 0;
  trigger: Trigger = new Trigger();
  triggerclass: string;
  userId: string;
  projectID: string;
  getTriggerGroup(): any {
    return this;
  }
}
